Make diffs relative from same directory

a16bitsysop opened the pull request:
and update for new version of postfix.

a16bitsysop committed:
Make diffs relative from same directory and update for new version of postfix.

a16bitsysop committed:
Update Readme.md

Change postfix version to new version and change hiredis path as should now be able install with a package

titusjose commented:
@a16bitsysop Thanks for the patch.

Since I have not been using this for a while, I do not have any setup handy to check if this patch compiles with postfix version that you have created diff.

Have you tested this patch?

titusjose commented on README.md:
```diff
@@ -1,14 +1,14 @@
 ### Postfix Redis
-This is a patch to the postfix 2.9.3 MTA to add a lookup table for redis database.
+This is a patch to the postfix 3 MTA to add a lookup table for redis database.
```
Can we be more specific of the postfix version here and not just major version.

a16bitsysop replied:
ok will have a look into it

titusjose commented on README.md:
```diff
@@ -1,14 +1,14 @@
 This patch depends on hiredis client library.
 
-Apply the patch to a vanilla postfix 2.9.3 source tree
+Apply the patch to a vanilla postfix 3 source tree
```
Can we be more specific of the postfix version here and not just major version.

titusjose commented on README.md:
```diff
@@ -1,14 +1,14 @@
+% make makefiles CCARGS="-DHAS_REDIS -I/usr/include/hiredis" AUXLIBS="-L/usr/lib -lhiredis"  
 Change the include and library location for hiredis client library if required.  
 
 ### Compile postfix  
```
Which distribution did you use? Would'nt the include directory change from distribution to distribution.

a16bitsysop replied:
I used alpine, but there is a package on ubuntu as well.  Will install it and check it installs to /usr or /usr/local

titusjose commented:
@a16bitsysop If you intent to use this in your project I would like to change this project to be dyncamically loaded database plugin for postfix 3 and above.

a16bitsysop commented:
Yes I am planning to, but just got to the getting it to compile (in alpine) stage.

a16bitsysop committed:
Combine patches as dynamicmaps.cf needs patch also
delete hiredis.h as it is in hiredis package

a16bitsysop commented:
Compiles postfix-redis.so but doesn’t install it, and not linked to libhiredis if ldd it.

a16bitsysop committed:
Patch some more files so postfix-redis.so installs with other plugins
Check which versions of postfix it works with

a16bitsysop committed:
But commands in code block style

a16bitsysop committed:
Add more code blocks

a16bitsysop commented:
compiles, links, and installs now.  Tested with postconf and postmap queries

titusjose commented:
@a16bitsysop Thanks for those quick changes. Sorry for the delays, I have not been able to focus on this because of the ongoing issues around here. I did glimpse through the patch. Just wanted to know if alpine ships the postfix package with dynamicmaps support like what debian and derivative do. For example, ubuntu latest version ships postfix with dynaicmaps and all supported database plugins as separate packages too.

a16bitsysop commented:
Yes they do the same, the main alpine packages are:

postfix-lmdb-3.4.9-r0
postfix-ldap-3.4.9-r0
postfix-mysql-3.4.9-r0
postfix-3.4.9-r0
postfix-stone-3.4.9-r0
postfix-pcre-3.4.9-r0
postfix-pgsql-3.4.9-r0
postfix-sqlite-3.4.9-r0

a16bitsysop committed:
Change layout to the same as dict_sqlite.c

a16bitsysop committed:
Add error message when cannot connect to the redis server

a16bitsysop committed:
Add lookup example

a16bitsysop committed:
add main.cf

a16bitsysop committed:
fix typo

a16bitsysop committed:
Change back to original style as dict_pgsql.c is the same
Add prefix in config parse to add a prefix to the key
Always print key looking up for debug purposes
Add main.cf and redis exmaples

a16bitsysop committed:
add $ for redis commands

a16bitsysop committed:
update readme

a16bitsysop committed:
put defaults in code block

a16bitsysop committed:
Add msg.h to Makefile.in for dict_redis.o
remove valid_hostname.h as not used

a16bitsysop committed:
fix typo

a16bitsysop committed:
fix heading size

a16bitsysop committed:
Add man page header

a16bitsysop committed:
add redis-cli monitor info to readme
make logging conditional again, uses postfix log verbosity
remake patch against 3.5.6

a16bitsysop committed:
fix typo in port number example

a16bitsysop committed:
add WIP readme and manpage

a16bitsysop committed:
WIP: add more documentation

a16bitsysop committed:
update documentation

a16bitsysop committed:
remove REDIS_README as can be generated from REDIS_README.html

a16bitsysop committed:
WIP: add upstream suggestions

a16bitsysop committed:
WIP: fix compilation error

a16bitsysop committed:
WIP: fix compilation error

a16bitsysop committed:
fix errors

a16bitsysop committed:
tidy up

a16bitsysop committed:
change message type

a16bitsysop committed:
fix spacing

a16bitsysop committed:
add upstream changes
move patches into own directory

a16bitsysop committed:
remove extra #endif

a16bitsysop committed:
add note about alpine linux
